test/system/252-quadlet.bats

```diff
@@ -0,0 +1,251 @@
+    STARTED_TIME=$(date "+%F %R:%S") # Start time for new log time
+}
+
+function setup() {
+    skip_if_remote "quadlet tests are meaningless over remote"
+
+    start_time
+
+    basic_setup
+}
+
+function teardown() {
+    for UNIT_FILE in ${UNIT_FILES[@]}; do
+        if [[ -e "$UNIT_FILE" ]]; then
+            local service=$(basename "$UNIT_FILE")
+            run systemctl stop "$service"
+            if [ $status -ne 0 ]; then
+               echo "# WARNING: systemctl stop failed in teardown: $output" >&3
+            fi
+            rm -f "$UNIT_FILE"
+        fi
+    done
+    systemctl daemon-reload
+
+    basic_teardown
+}
+
+# Converts the quadlet file and installs the result it in $UNIT_DIR
+function run_quadlet() {
+    local sourcefile="$1"
+    local service=$(quadlet_to_service_name "$sourcefile")
+
+    # quadlet always works on an entire directory, so copy the file
+    # to transform to a tmpdir
+    local quadlet_tmpdir=$(mktemp -d --tmpdir=$PODMAN_TMPDIR quadlet.XXXXXX)
+    cp $sourcefile $quadlet_tmpdir/
+
+    QUADLET_UNIT_DIRS="$quadlet_tmpdir" run $QUADLET $_DASHUSER $UNIT_DIR
+    assert $status -eq 0 "Failed to convert quadlet file: $sourcefile"
+    is "$output" "" "quadlet should report no errors"
+
+    # Ensure this is teared down
+    UNIT_FILES+=("$UNIT_DIR/$service")
+
+    QUADLET_SERVICE_NAME="$service"
+
+    cat $UNIT_DIR/$QUADLET_SERVICE_NAME
+}
+
+function service_setup() {
+    local service="$1"
+
+    systemctl daemon-reload
+
+    run systemctl start "$service"
+    assert $status -eq 0 "Error starting systemd unit $service: $output"
+
+    run systemctl status "$service"
+    assert $status -eq 0 "systemctl status $service: $output"
+}
+
+# Helper to stop a systemd service running a container
+function service_cleanup() {
+    local service="$1"
+    local expected_state="$2"
+
+    run systemctl stop "$service"
+    assert $status -eq 0 "Error stopping systemd unit $service: $output"
+
+    # Regression test for #11304: confirm that unit stops into correct state
+    if [[ -n "$expected_state" ]]; then
+        run systemctl show --property=ActiveState "$service"
+        assert "$output" = "ActiveState=$expected_state" \
+               "state of service after systemctl stop"
+    fi
+
+    rm -f "$UNIT_DIR/$service"
+    systemctl daemon-reload
+}
+
+@test "quadlet - basic" {
+    local quadlet_file=$PODMAN_TMPDIR/basic_$(random_string).container
+    cat > $quadlet_file <<EOF
+[Container]
+Image=$IMAGE
+Exec=top
+EOF
+
+    run_quadlet "$quadlet_file"
+    service_setup $QUADLET_SERVICE_NAME
+
+    local container_name=systemd-$(basename $quadlet_file .container)
+
+    # Give container time to start; make sure output looks top-like
+    sleep 2
+    run journalctl "--since=$STARTED_TIME" --unit="$QUADLET_SERVICE_NAME"
+    is "$output" '.*Load average.*'
+
+    run_podman container inspect  --format "{{.Id}}" $container_name
+    assert $status -eq 0 "Error inspecting container"
```

Comment:
No need for the $status check when using `run_podman`.  If you expect a non-zero exit code, you can use `run_podman 125 ...` for instance.

I suggest checking the output of `inspect` instead.  For instance, we could check whether the container is running via `--format "{{.State.Status}}"` and expect "running" via `is "$output" "running" "container should be started by systemd and hence be running"`.